sale_commission/model/invoice.py

```diff
@@ -0,0 +1,192 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2011 Pexego Sistemas Informáticos (<http://www.pexego.es>).
+#    All Rights Reserved
+#    $Id$
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+"""invoice agents"""
+
+from openerp import models, fields, api, _
+
+
+class invoice_line_agent(models.Model):
+    """invoice agents"""
+
+    _name = "invoice.line.agent"
+
+    invoice_line_id = fields.Many2one(
+        "account.invoice.line",
+        string="Invoice Line",
+        required=True,
+        ondelete="cascade"
+    )
+
+    invoice_id = fields.Many2one(
+        "account.invoice",
+        string="Invoice",
+        related="invoice_line_id.invoice_id"
+    )
+
+    invoice_date = fields.Date(
+        string="Invoice date",
+        related="invoice_id.date_invoice",
+        readonly=True
+    )
+
+    agent_id = fields.Many2one(
+        "sale.agent",
+        string="Agent",
+        required=True,
+        ondelete="cascade"
+    )
+
+    commission_id = fields.Many2one(
+        "commission",
+        string="Applied commission",
+        required=True,
+        ondelete="cascade",
+    )
+
+    settled = fields.Boolean(
+        string="Settled",
```

Comment:
Just a non blocking tip, string is not mandatory here.
*string – the label of the field seen by users (string); if not set, the ORM takes the field name in the class (capitalized).* 
https://www.odoo.com/documentation/8.0/reference/orm.html#fields